cmd/kubeadm/app/cmd/phases/etcd.go

```diff
@@ -19,84 +19,71 @@ package phases
 import (
 	"fmt"
 
-	"github.com/spf13/cobra"
-
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
-	kubeadmscheme "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme"
-	kubeadmapiv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
-	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
-	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
 	etcdphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/etcd"
 	"k8s.io/kubernetes/pkg/util/normalizer"
 )
 
 var (
-	etcdLocalLongDesc = fmt.Sprintf(normalizer.LongDesc(`
-		Generates the static Pod manifest file for a local, single-node etcd instance and saves it to %s file.
-		`+cmdutil.AlphaDisclaimer), kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.Etcd, kubeadmconstants.GetStaticPodDirectory()))
-
 	etcdLocalExample = normalizer.Examples(`
 		# Generates the static Pod manifest file for etcd, functionally 
-		# equivalent to what generated by kubeadm init.
-		kubeadm alpha phase etcd local
+		# equivalent to what is generated by kubeadm init.
+		kubeadm init phase etcd local
 
-		#  Generates the static Pod manifest file for etcd.
-		kubeadm alpha phase etcd local --config masterconfiguration.yaml
+		# Generates the static Pod manifest file for etcd using options
+		# read from a configuration file.
+		kubeadm init phase etcd local --config config.yaml
 		`)
 )
 
-// NewCmdEtcd returns main command for Etcd phase
-func NewCmdEtcd() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "etcd",
-		Short: "Generates static Pod manifest file for etcd.",
-		Long:  cmdutil.MacroCommandLongDescription,
-	}
-
-	manifestPath := kubeadmconstants.GetStaticPodDirectory()
-	cmd.AddCommand(getEtcdSubCommands(manifestPath, "")...)
-	return cmd
+type etcdData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	ManifestDir() string
 }
 
-// getEtcdSubCommands returns sub commands for etcd phase
-func getEtcdSubCommands(outDir, defaultKubernetesVersion string) []*cobra.Command {
-
-	cfg := &kubeadmapiv1beta1.InitConfiguration{}
-
-	// Default values for the cobra help text
-	kubeadmscheme.Scheme.Default(cfg)
-
-	var cfgPath string
-	var subCmds []*cobra.Command
-
-	properties := struct {
-		use      string
-		short    string
-		long     string
-		examples string
-		cmdFunc  func(outDir string, cfg *kubeadmapi.InitConfiguration) error
-	}{
-		use:      "local",
-		short:    "Generates the static Pod manifest file for a local, single-node etcd instance",
-		long:     etcdLocalLongDesc,
-		examples: etcdLocalExample,
-		cmdFunc:  etcdphase.CreateLocalEtcdStaticPodManifestFile,
+// NewEtcdPhase creates a kubeadm workflow phase that implements handling of etcd.
+func NewEtcdPhase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:    "etcd",
+		Short:   "Generates static Pod manifest file for local/stacked etcd.",
```

Comment:
I don't think you need to say "stacked" in multiple locations, I'd probably remove that.